guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -290,12 +295,15 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
                                 this.state.currentNode.serialize(serializationObject);
                                 const newControl = Control.Parse(serializationObject, this.props.globalState.guiTexture);
 
-                                if (newControl) { //insert the new control into the adt
+                                if (newControl) { //insert the new control into the adt or parent container
                                     this.props.globalState.workbench.appendBlock(newControl);
+                                    this.props.globalState.guiTexture.removeControl(newControl);
+                                    this.state.currentNode.parent?.addControl(newControl);
+
                                     let index = 1;
-                                    while (this.props.globalState.workbench.nodes.filter(  //search if there are any copies
+                                    while ( this.props.globalState.guiTexture.getDescendants(false).filter(  //search if there are any copies

```

Comment:
why does this need to be iterated in a while loop? why don't you filter all duplicates and change their names?

Reply:
Because people will only be copying one at a time. there may also be a "item copy 2" or and "item copy 5?" Now we could filter names out by searching for "item" but then we can get cases like "item item copy" or "item 2"

Reply:
so you are adding the item with the same name, iterate to check if there are 2 items with the same name, and then change the name, until it is no longer a duplicate?

Wouldn't iterating over `${this.state.currentNode.name} Copy ${index++}` until it is not found be enough?